wordpress-mobile/WordPress-iOS
# WeeklyRoundupBackgroundTask - Change to use BGProcessingTask

I've been able to re-create `BAD_EXC_ACCESS` exception only when setting a breakpoint and the WeeklyRoundupBackgroundTask triggers and intentionally leaving the breakpoint hanging for a very long time.

This PR changes the `WeeklyRoundupBackgroundTask` to use [BGProcessingTask](https://developer.apple.com/documentation/backgroundtasks/bgprocessingtask)  instead of `BGAppRefreshTask` on the guess that this exception is caused by iOS running out of processing time.  I have not been able to re-create this exception when running the app on device normally i.e. in non debug session

I have wrapped this change in a new feature flag 'weeklyRoundupBGProcessingTask'

Ref #17956

To test:
1. Test on device.  When app launches go to Me -> App Settings -> Debug.  Tap Weekly Roundup
2. If you don't have enough blogs / history also slide to the right to "Include A8c P2s"
3. Tap one of the options ie immediately and then minimise the app in background
Weekly Round up Notification should eventually appear as normal.

Additionally I've added logging to confirm that the old style backgroundTask is being ended correctly.   The iOS warning stating that background task still not ended is output when the expirationHandler is triggered.  To test this:
1. Run the app 
2. Minimise the app in background 
3. Relaunch the app 
4. Filter on BackgroundTask like in screenshot below.  Example screenshot showing that BGTaskID 15 is being correctly ended.

![image](https://user-images.githubusercontent.com/88816724/155713365-0443e304-6cec-4015-b4e7-e98e34cc7b20.png)



1. Potential unintended areas of impact
WeeklyRoundupBackgroundTask

6. What I did to test those areas of impact (or what existing automated tests I relied on)
I tested on device that the push notification would still send

7. What automated tests I added (or what prevented me from doing so)
None

PR submission checklist:
